contrib/scripts/env-setup.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python
+import argparse
+import collections
+from datetime import datetime, timedelta
+import os
+import re
+import requests
+import subprocess
+import threading
+
+
+VersionConfig = collections.namedtuple('VersionConfig', ['wheels', 'tpu'])
+STABLE_VERSION_MAP = {
+  'xrt==1.15.0': VersionConfig('1.15', '1.15.0'),
+}
+DIST_BUCKET = 'gs://tpu-pytorch/wheels'
+TORCH_WHEEL_TMPL = 'torch-{whl_version}-cp36-cp36m-linux_x86_64.whl'
+TORCH_XLA_WHEEL_TMPL = 'torch_xla-{whl_version}-cp36-cp36m-linux_x86_64.whl'
+TORCHVISION_WHEEL_TMPL = 'torchvision-{whl_version}-cp36-cp36m-linux_x86_64.whl'
+
+
+def update_tpu_runtime(tpu_ip, version):
+  print('Updating TPU runtime to {} ...'.format(args.version))
+  url = 'http://{tpu_ip}:8475/requestversion/{tpu_version}'.format(
+    tpu_ip=tpu_ip, tpu_version=version.tpu)
+  print('Done updating TPU runtime: {}'.format(requests.post(url)))
+
+def get_version(version):
+  if version == 'nightly':
+    return VersionConfig('nightly', 'TPU-dev{}'.format(
+      (datetime.today() - timedelta(1)).strftime('%Y%m%d')))
+
+  try:
+    datetime.datetime.strptime(version, '%Y%m%d')
+    return VersionConfig(f'nightly+{version}', f'TPU-dev{version}')
+  except ValueError:
+    pass  # Not a dated nightly.
+
+  if version not in VERSION_MAP:
+    raise ValueError(f'Version {version} unknown')
+  return VERSION_MAP[version]
+
+def parse_env_tpu_ip():
+  # In both Colab and Kaggle: TPU_NAME='grpc://abc.def.ghi.jkl:8470'
+  tpu_addr = os.environ.get('TPU_NAME', None)
+  tpu_ip_regex = re.compile('grpc://(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}):8470')
+  m_tpu_ip = tpu_ip_regex.match(tpu_addr)
+  if not m_tpu_ip:
+    raise ValueError('TPU not found.')
+
+  return m_tpu_ip.group(1)
+
+def install_vm(version, apt_packages):
+  torch_whl = TORCH_WHEEL_TMPL.format(whl_version=version.wheels)
+  torch_whl_path = os.path.join(DIST_BUCKET, torch_whl)
+  torch_xla_whl = TORCH_XLA_WHEEL_TMPL.format(whl_version=version.wheels)
+  torch_xla_whl_path = os.path.join(DIST_BUCKET, torch_xla_whl)
+  torchvision_whl = TORCHVISION_WHEEL_TMPL.format(whl_version=version.wheels)
+  torchvision_whl_path = os.path.join(DIST_BUCKET, torchvision_whl)
+  apt_pkgs = ' '.join(apt_packages)
+
+  installation_cmds = [
+    'pip uninstall -y torch torchvision',
+    f'gsutil cp {torch_whl_path} .',
+    f'gsutil cp {torch_xla_whl_path} .',
+    f'gsutil cp {torchvision_whl_path} .',
+    f'pip install {torch_whl}',
+    f'pip install {torch_xla_whl}',
+    f'pip install {torchvision_whl}',
+    f'sudo apt-get install {apt_pkgs}',
+  ]
+  for cmd in installation_cmds:
+    subprocess.call(cmd.split())
```

Comment:
I usually rely on shlex.split but I guess these commands are simple enough this would work.

Reply:
Made them a list as Davide suggested.